jsk_arc2017_baxter/euslisp/lib/stow-interface.l

```diff
@@ -0,0 +1,183 @@
+    (send self :delete-tote-scene arm)
+    (send self :pick-object-in-tote arm
+          :n-trial 2
+          :n-trial-same-pos 1
+          :do-stop-grasp nil)
+    (send *ri* :angle-vector
+          (send self :ik->tote-center arm
+                :offset #f(0 0 300) :rotation-axis :z :use-gripper t)
+          3000 (send *ri* :get-arm-controller arm) 0)
+    (send *ri* :wait-interpolation)
+    (setq graspingp (send *ri* :graspingp arm))
+    (ros::ros-info "[main] arm: ~a graspingp: ~a" arm graspingp)
+    graspingp)
+  (:return-from-pick-object (arm)
+    (send *ri* :stop-grasp arm)
+    (ros::ros-info "[main] arm: ~a return from pick-object to fold-pose-back" arm)
+    (send *ri* :angle-vector-sequence
+          (list (send *baxter* :avoid-shelf-pose arm (if (eq arm :larm) :d :f))
+                (send *baxter* :fold-pose-back arm))
+          :fast (send *ri* :get-arm-controller arm) 0 :scale 5.0)
+    (send *ri* :wait-interpolation))
+  (:verify-object (arm)
+    (setq picking-fail-count 0)
+    ;; TODO: in-hand verification
+    ;; TODO: (setq target-obj (send *ti* :recognize-object-in-hand arm))
+    ;; TODO: (return-from :verify-objec target-obj)
+    (setq target-obj t)
+    target-obj)
+  (:check-picking-fail-count (arm)
+    (incf picking-fail-count)
+    (ros::ros-info "[main] arm: ~a, picking fail count: ~a" arm picking-fail-count)
+    (> picking-fail-count 1))
+  (:set-target-bin (arm)
+    ;; TODO: when we select target-bin, we should consider target-obj size.
+    ;; TODO: (setq target-bin (send *ti* :get-target-bin arm target-obj))
+    (setq target-bin
+          (if (eq arm :larm) :a (elt (list :b :c) (random 2))))
+    (if (eq target-bin :tote)
+      (progn
+        (ros::ros-info "[main] unable to recognize, return to tote")
+        nil)
+      (progn
+        (ros::set-param (format nil "~a_hand/target_bin" (arm2str arm)) (symbol2str target-bin))
+        (ros::ros-info "[main] target-bin: ~a" target-bin)
+        t)))
+  (:return-object (arm)
+    (send *ri* :angle-vector
+          (send self :ik->tote-center arm :offset #f(0 0 0)
+                :rotation-axis :z :use-gripper t)
+          :fast (send *ri* :get-arm-controller arm) 0 :scale 3.0)
+    (send *ri* :wait-interpolation)
+    (send *ri* :stop-grasp arm)
+    (send self :spin-off-by-wrist 5)
+    (send *ri* :wait-interpolation)
+    (ros::ros-info "[main] ~a, return object in tote" arm)
+    (send self :add-tote-scene arm)
+    (send *ri* :fold-pose-back arm)
+    (send *ri* :wait-interpolation))
+  (:place-object (arm)
+    (let (dropped)
+      (ros::ros-info "[main] ~a, place object in bin ~a" arm target-bin)
+      (send self :add-tote-scene arm)
+      (send *baxter* :head_pan :joint-angle (if (eq arm :larm) 80 -80))
+      (send *ri* :angle-vector
+            (send self :ik->bin-center arm target-bin
+                  :offset #f(0 0 300) :rotation-axis :z :use-gripper t)
+            4000 (send *ri* :get-arm-controller arm) 0)
```

Comment:
currently, `scale` is not supported by `moveit` method.
if you set tm as `:fast`, `angle-vector-raw` will be executed instead.